src/plugins/mer/merplugin.cpp

```diff
@@ -409,5 +412,60 @@ void MerPlugin::ensureCustomRunConfigurationIsTheDefaultOnCompilationDatabasePro
     });
 }
 
+void MerPlugin::ensureDailyUpdateCheckForEarlyAccess()
+{
+    QInstaller::Repository::registerMetaType();
+    QDir iniDir = QCoreApplication::applicationDirPath();
+    iniDir.cdUp();
+    if (Utils::HostOsInfo::isMacHost()) {
+        iniDir.cdUp();
+        iniDir.cdUp();
+        iniDir.cdUp();
+    }
+    QSettings settings(iniDir.filePath("SDKMaintenanceTool.ini"), QSettings::IniFormat);
+    const QVariantList variants = settings.value(QLatin1String("DefaultRepositories"))
+        .toList();
+    foreach (const QVariant &variant, variants) {
+        QInstaller::Repository repo = variant.value<QInstaller::Repository>();
+        if (repo.m_enabled && repo.m_url.toString().endsWith("updates-ea")) {
+            QSettings *settings = ICore::settings();
+            settings->beginGroup(QLatin1String("Updater"));
+            settings->setValue(QLatin1String("CheckUpdateInterval"), QLatin1String("DailyCheck"));
+            settings->endGroup();
+        }
+    }
```

Comment:
I would add a safety QTC_CHECK that (e.g.) the "updates-ea" repo is in the list, i.e., that we successfully parsed the .ini, which may change in future.

I would consider reverting to weekly check when EA updates are disabled later.

Reply:
QTC_CHECK sounds like a good idea.

I thought about reverting back to weekly, but then decided against it: We would have to store the information that the change to daily was done by us, and not by the user. We don't want to change the setting if the user has modified it, unless the user has enabled the EA repo, in which case we want to force it to daily. I don't find the extra effort in storing the information about us modifying the setting worth it.

Reply:
Yeah, I thought it was like that :)